launchable/commands/inspect/tests.py

```diff
@@ -1,23 +1,154 @@
+import json
 import os
 import sys
+from abc import ABCMeta, abstractmethod
 from http import HTTPStatus
+from typing import List
 
 import click
 from tabulate import tabulate
 
 from ...utils.env_keys import REPORT_ERROR_KEY
 from ...utils.launchable_client import LaunchableClient
+from ...utils.session import parse_session
+from ..helper import require_session
+
+
+class TestResult(object):
+    def __init__(self, result: dict):
+        self._status = result.get("status", "")
+        self._duration_sec = result.get("duration", 0.0)
+        self._created_at = result.get("createdAt", None)
+        self._test_path = "#".join([path["type"] + "=" + path["name"]
+                                   for path in result["testPath"] if path.keys() >= {"type", "name"}])
+
+
+class TestResults(object):
+    def __init__(self, results: List[TestResult]):
+        self._results = results
+
+    def add(self, result: TestResult):
+        self._results.append(result)
+
+    def list(self) -> List[TestResult]:
+        return self._results
+
+    def total_duration_sec(self) -> float:
+        return sum([result._duration_sec for result in self._results])
+
+    def total_duration_min(self) -> float:
+        return (sum([result._duration_sec for result in self._results]) / 60)
+
+    def total_count(self) -> int:
+        return len(self._results)
+
+    def filter_by_status(self, status: str) -> 'TestResults':
+        return TestResults([result for result in self._results if result._status == status])
+
+
+class TestResultAbstractDisplay(metaclass=ABCMeta):
+    def __init__(self, results: TestResults):
+        self._results = results
+
+    @abstractmethod
+    def display(self):
+        raise NotImplementedError("display method is not implemented")
+
+
+class TestResultJSONDisplay(TestResultAbstractDisplay):
+    def __init__(self, results: TestResults):
+        super().__init__(results)
+
+    def display(self):
+        result_json = {}
+        result_json["summary"] = {
+            "total": {
+                "report_count": self._results.total_count(),
+                "duration_min": round(self._results.total_duration_min(), 2),
```

Comment:
How did you decide that rounding a number to only two decimals is enough?

Reply:
Aligned it with the table layout  https://github.com/launchableinc/cli/pull/801/files#diff-c9d4de19c85826febc29e1f0427814a89796f16e4968205aaeb585e967c357e8L51 and because if left as is, discrepancies can occur due to different versions https://github.com/launchableinc/cli/actions/runs/8182233199/job/22373269353#step:10:116. 
That's it and there's no other reason

Reply:
Ok, thank you.